core/src/main/java/org/apache/iceberg/BaseFileScanTask.java

```diff
@@ -100,6 +140,20 @@ public long length() {
       return len;
     }
 
+    @Override
+    public long sizeBytes() {
+      if (sizeBytes == Long.MIN_VALUE) {
+        this.sizeBytes = FileScanTask.super.sizeBytes();
```

Comment:
This would only be called after serialization. We init `sizeBytes` in the constructor otherwise.